core/amber/src/main/scala/edu/uci/ics/texera/web/resource/auth/UserResource.scala

```diff
@@ -49,29 +49,45 @@ class UserResource {
   @Path("/login")
   def login(@Session session: HttpSession, request: UserLoginRequest): Response = {
 
-    // try to fetch the record
-    val user = this.userDao.fetchOneByName(request.userName)
-    if (user == null) { // not found
+    // try to fetch the UID given the username and password
+    val userUid = SqlServer.createDSLContext
+      .select(USER.UID)
+      .from(USER)
+      .where(
+        USER.NAME
+          .eq(request.userName)
+          .and(USER.PASSWORD.eq(PasswordEncryption.EncoderByMd5(request.password)))
```

Comment:
Using MD5 as a password hashing function is not ideal: it's too fast to compute. Try to follow this example to do hashing with `org.jasypt.util.password.StrongPasswordEncryptor`.